### ISSUE TITLE
MeasurementSpecification in NEGOTIATE_ALGORITHM

### ISSUE THREAD

**Issue by hiption** [DMTF/libspdm#1423]
Hi all,
I have a question about MeasurementSpecification field in NEGOTIATE_ALGORITHM.

https://github.com/DMTF/libspdm/blob/dc93a39396efaf083815d57c46abc92fd063e568/library/spdm_requester_lib/libspdm_req_negotiate_algorithms.c#L303-L317

According to above code, the requester returns INVALID_MSG_FIELD status as follows;
1. Responder has a MEAS_CAP.
2. Requester does not want to use measurement.
3. Requester sets MeasurementSpecification as zero in NEGOTIATE_ALGORITHM.
4. Responder sets MeasurementSpecificationSel as zero in ALGORITMS.
5. Requester returns INVALID_MSG_FIELD status because the MeasurementSpecificationSel is not SPDM_MEASUREMENT_BLOCK_HEADER_SPECIFICATION_DMTF, but the Responder has a MEAS_CAP.

In some case, even if the responder has the measurement capability, the requester may not want to use measurement.

**Comment by steven-bellock** [comment]
I thought this was fixed in https://github.com/DMTF/libspdm/pull/876 but that looks like it was all on the Responder side.

**Comment by hiption** [comment]
BTW, how to access [https://github.com/DMTF/Security-TF](https://github.com/DMTF/Security-TF/issues/2014)?
Can only DMTF members access the Security-TF?

**Comment by steven-bellock** [comment]
> Can only DMTF members access the Security-TF?

Correct. If you are not a DMTF member and have questions / issues about the specification you can ask on the public http://spdmforum.com/.

**Comment by hiption** [comment]
I think one more condition should be added as follows;

if (libspdm_is_capabilities_flag_supported( 
         spdm_context, true, 0, 
         SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP)
         && (spdm_request->measurement_specification != 0)) {
